crates/bitwarden/src/vault/cipher/cipher.rs

```diff
@@ -366,23 +368,49 @@ impl CipherView {
         }
     }
 
+    fn reencrypt_attachment_keys(
+        &mut self,
+        old_key: &SymmetricCryptoKey,
+        new_key: &SymmetricCryptoKey,
+    ) -> Result<()> {
+        // If the cipher does not have a key, we need to reencrypt all attachment keys
+        if let Some(attachments) = &mut self.attachments {
+            // If any attachment is missing a key we can't move the cipher
+            if attachments.iter().any(|a| a.key.is_none()) {
+                return Err("This cipher contains attachments without keys. Those attachments need to be reuploaded before the cipher can be moved".into());
+            }
+
+            for attachment in attachments {
+                let attachment_key = attachment
+                    .key
+                    .as_mut()
+                    .expect("Key is asserted to be present");
+                let dec_attachment_key: Vec<u8> = attachment_key.decrypt_with_key(old_key)?;
+                *attachment_key = dec_attachment_key.encrypt_with_key(new_key)?;
```

Comment:
Should we expose a helper on the `AttachmentView` instead?

Reply:
My thinking here was that we always want to reencrypt all the attachments in one go or return an error, which is also why I'm checking if all the attachments have a key before doing the reencryption.

If we exposed this as a helper in `AttachmentView` it would be easier to end up in a situation where we reencrypt one attachment but not another.

That said I think we're doing reencryption in enough places that it would be convenient to have some generic helper functions for it, and that would also simplify this code a bit more.